src/test/java/com/switchfully/parkshark_2024_10/user/UserRepositoryTest.java

```diff
@@ -0,0 +1,82 @@
+package com.switchfully.parkshark_2024_10.user;
+
+import com.switchfully.parkshark_2024_10.division.Division;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+public class UserRepositoryTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    void findByEmailAndPassword_shouldReturnCorrectPerson() {
+        // given
+        Division division = new Division();
+
+        Director director = new Director(division, "Jane", "Doe", "jane.doe@parkshark.com", "password123");
+        userRepository.save(director);
+
+        // When
+        Person found = userRepository.findByEmailAndPassword("jane.doe@parkshark.com", "password123");
```

Comment:
An issue here is that we are requesting the user by its email address (which is fine) and its unencoded password (which is not). The repository should probably not know how the password is stored and/or encoded, it just sends the data through to the DataSource as provided. The code here should, in return, fail because it should not find the user by this combination.

We should most likely encode the password before sending it through to the Repository, because it should've been stored in the datasource as the encoded value, not the human-readable one.

Reply:
fair point, fixed it in the test